refactor repeating code

batuhan opened the pull request.

bcoe commented:
@batuhan I really like the cleanup of pushing the providers into a list :+1: 

I usually try to make sure that any tools I write work all the way back to Node 0.10, since folks installing the tool who aren't Node affianodos are often running older platforms.

tldr; perhaps update this patch to include your refactor without the ES2015 constructs.

batuhan commented:
I've updated the es6 branch to not include es6 :) 

So, just added the checker generator thingy.

There seems to be problem with the whois fallback with my updates. I'm checking it right now.

bcoe commented:
@batuhan looks good to me, coverage just dropped because you reduced the lines of code with cleanup.

any reason not to land this?
